EssentialApp/EssentialAppTests/CommentsUIIntegrationTests.swift

```diff
@@ -0,0 +1,199 @@
+		let table = "Feed"
+		let bundle = Bundle(for: CommentsPresenter.self)
+		let value = bundle.localizedString(forKey: key, value: nil, table: table)
+		if value == key {
+			XCTFail("Missing localized string for key: \(key) in table: \(table)", file: file, line: line)
+		}
+		return value
+	}
+	
+	private func assert(_ sut: CommentsController, isRendering comments: [Comment], file: StaticString = #file, line: UInt = #line) {
+		guard sut.numberOfRenderedCommentViews == comments.count else {
+			return XCTFail("Expected \(comments.count) comments, got \(sut.numberOfRenderedCommentViews) instead", file: file, line: line)
+		}
+		comments.enumerated().forEach { index, comment in
+			assertThat(sut, hasViewConfiguredFor: comment, at: index, file: file, line: line)
+		}
+	}
+	
+	private func assertThat(_ sut: CommentsController, hasViewConfiguredFor comment: Comment, at index: Int, file: StaticString = #file, line: UInt = #line) {
+		let view = sut.CommentView(at: index)
+		
+		guard let cell = view as? CommentCell else {
+			return XCTFail("Expected \(CommentCell.self) instance, got \(String(describing: view)) instead", file: file, line: line)
+		}
+		
+		let formatter = RelativeDateTimeFormatter()
+		let relativeDate = formatter.localizedString(for: comment.createdAt, relativeTo: Date())
+		
+		XCTAssertEqual(cell.author, comment.author, "Expected author to be \(comment.author) for comment at index \(index)", file: file, line: line)
+		XCTAssertEqual(cell.date, relativeDate, "Expected date to be \(comment.createdAt) for comment at index \(index)", file: file, line: line)
+		XCTAssertEqual(cell.comment, comment.message, "Expected message to be \(comment.message) for comment at index \(index)", file: file, line: line)
+	}
+	
+	private func makeComment(id: UUID = UUID(), message: String = "message", createdAt: Date = Date(), author: String = "author") -> Comment {
+		return Comment(id: id, message: message, createdAt: createdAt, author: author)
+	}
+	
+	class LoaderSpy: CommentsLoader {
+		private var completions = [(CommentsLoader.Result) -> Void]()
+		var loadCallCount: Int {
+			return completions.count
+		}
+		
+		func load(completion: @escaping (CommentsLoader.Result) -> Void) {
+			completions.append(completion)
+		}
+		
+		func completeCommentsLoading(with comments: [Comment] = [], at: Int) {
+			completions[at](.success(comments))
+		}
+		
+		func completeCommentsLoadingWithError(at: Int) {
+			let error = NSError(domain: "an error", code: 0)
+			completions[at](.failure(error))
+		}
+	}
+}
+
+private extension CommentsController {
+	func simulateUserInitiatedCommentsReload() {
+		refreshControl?.simulatePullToRefresh()
+	}
+	
+	var isShowingLoadingIndicator: Bool {
+		return refreshControl?.isRefreshing == true
+	}
+	
+	var numberOfRenderedCommentViews: Int {
+		return tableView.numberOfRows(inSection: 0)
+	}
+	
+	func CommentView(at index: Int) -> UITableViewCell? {
```

Comment:
the convention is to start method names with lowercase letters

```suggestion
	func commentView(at index: Int) -> UITableViewCell? {
```